src/test/java/com/pnu/dev/shpaltaif/PublicAccountServiceIntegrationTest.java

```diff
@@ -134,64 +129,108 @@ public void createAndThenReadAndThenUpdateAndThenDeleteSuccessFlow() {
     }
 
     @Test
-    @Transactional
-    public void updatePseudonymTest() {
+    public void updateSignatureTest() {
+
+        //Create PublicAccount
+        PublicAccount publicAccount = createPublicAccount("name", "surname");
+        PublicAccount publicAccountFromDb = publicAccountService.findById(publicAccount.getId());
+
+        //Test default signature
+        String expectedSignature = "name surname";
+        assertEquals(expectedSignature, publicAccountFromDb.getSignature());
+
+        //Set pseudonym
+        String pseudonym = " pseudonym ";
+        PublicAccountDto publicAccountDto = PublicAccountDto.builder()
+                .name(publicAccountFromDb.getName())
+                .surname(publicAccountFromDb.getSurname())
+                .pseudonymUsed(false)
+                .pseudonym(pseudonym)
+                .build();
+        publicAccountService.update(publicAccountDto, publicAccountFromDb.getId());
 
-        PublicAccount publicAccount1 = createUserWriter("username1").getPublicAccount();
+        //Test signature when pseudonym has been set, but pseudonym is disabled
+        publicAccountFromDb = publicAccountService.findById(publicAccount.getId());
+        assertEquals(expectedSignature, publicAccountFromDb.getSignature());
 
-        PublicAccountDto publicAccountDtoWithoutPseudonym = PublicAccountDto.builder()
-                .name(publicAccount1.getName())
-                .surname(publicAccount1.getSurname())
+        //Activate pseudonym usage
+        publicAccountDto = PublicAccountDto.builder()
+                .name(publicAccountFromDb.getName())
+                .surname(publicAccountFromDb.getSurname())
                 .pseudonymUsed(true)
+                .pseudonym(pseudonym)
                 .build();
+        publicAccountService.update(publicAccountDto, publicAccountFromDb.getId());
 
-        ServiceException thrown = assertThrows(ServiceException.class,
-                () -> publicAccountService.update(publicAccountDtoWithoutPseudonym, publicAccount1.getId()));
-        assertEquals(thrown.getMessage(), "Щоб використовувати псевдонім, введіть його коректно");
+        //Test signature when pseudonym has been set and pseudonym is active
+        publicAccountFromDb = publicAccountService.findById(publicAccount.getId());
+        expectedSignature = pseudonym.trim();
```

Comment:
Do not reassign variables, it's really become hard to read.
Just create a new variable each time.
Or use value directly in assertEquals(), because it's clear enough, that first arg have to be an expected result.